RyftCore/Source/Models/Money.swift

```diff
@@ -0,0 +1,76 @@
+import Foundation
+
+public struct Money: Equatable, Hashable {
+
+    public let currency: Currency
+    public let amount: Int
+
+    public init(currency: Currency, amount: Int) {
+        self.currency = currency
+        self.amount = amount
+    }
+
+    public init(currencyCode: String, amount: Int) {
+        self.currency = Currency.from(code: currencyCode)
+        self.amount = amount
+    }
+
+    public func calculatePercentageAmount(with percentage: Double) -> Money {
+        let intermediateResult = calculatePercentagePreservingDecimal(percentage: percentage)
+        return Money(currency: self.currency, amount: Int(intermediateResult))
+    }
+
+    public func amountInMajorUnits() -> NSDecimalNumber {
```

Comment:
will be used to construct the amount required for ApplePay (within `PKPaymentRequest`) as that wants it in major units